### PR TITLE
#CP-3 Setup Eslint Prettier and Husky

### PR DESCRIPTION
**Pull request by Yvad60**
#### What does this PR do?

- [x] Setup and configure Eslint to use Airbnb
- [x] Setup and configure Prettier
- [x] Setup and configure Husky to run all scripts before committing 

#### Description of Task to be completed
- installing and configuring Eslint for code analysis and to quickly find problems using Airbnb style guide 
- installing and configuring Prettier as a code formatter 
- installing and configuring Husky to run all scripts and ensure all processes are in order before committing

#### How should this be manually tested?
Clone the project [repository](https://github.com/atlp-rwanda/phantom-fe-codebandits)

```
cd phantom-fe-codebandits/
git checkout ch-setup-eslint-prettier-husky-cp-3
git fetch origin
npm install
```
#### Any background context you want to provide?
- install Eslint and Prettier VS code extensions (optional but recommended)
#### What are relevant Trello cards?
[#CP-3](https://trello.com/c/DuGNLuvG/3-setup-eslint-prettier-husky)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,16 @@
   "type": "module",
   "scripts": {
     "start": "webpack serve",
-    "build": "webpack"
+    "build": "webpack",
```
Comment (Jkadhuwa) [inline package.json:9]: Could you add scripts for liniting and formatting

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "ecmaVersion": "latest",
+    "sourceType": "module"
+  },
+  "plugins": ["react"],
+  "rules": {
+    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }]
```
Comment (Jkadhuwa) [inline .eslintrc.json:20]: Add a rule to allow Arrow functions, this will come handy when using functional components.

**Review by Jkadhuwa — CHANGES_REQUESTED** [review]
Nice work, work on the feedback below

File: .eslintrc.json (hunk 2 above)
Comment (Yvad60) [reply .eslintrc.json:20]: are aI don't quite get that, are arrow functions blocked in Airbnb style guide?

**Review by Jkadhuwa — APPROVED** [review]

**Pull request merged**